File: src/RecordsManagmentWeb/ClientApp/src/app/requests-table/request.ts
import { RequestStatusEnum } from './request-status';

export class RequestDoc {
    id: number;
    requestId: string;
    type: number;
    claimNumber: string;
    insuredName: string;
    created: string;
    amount: number;
    dateOfLoss: string;
    dateOfService: string;
    phase: string;
    deatilsFileName: string;
    esRef?: number;
    status: RequestStatusEnum;
}

File: src/Models/Request.cs
using System;

namespace Models
{
    public class Request
    {
        public int Id { get; set; }
        public string RequestId { get; set; }
        public RequestTypeEnum Type { get; set; }
        public string ClaimNumber { get; set; }
        public string InsuredName { get; set; }
        public DateTime Created { get; set; }
        public decimal Amount { get; set; }
        public DateTime DateOfLoss { get; set; }
        public DateTime DateOfService { get; set; }
        public string Phase { get; set; }
        public string DeatilsFileName { get; set; }        
        public string PdfPackName { get; set; }
        public int? EsRef { get; set; }
        public RequestStatusEnum Status { get; set; }
        public bool Archived { get; set; }
        public bool Duplicated { get; set; }


        public string GetZipFileName() => $"{ClaimNumber}_{RequestId}.zip";
        
    }
}
